src/vehicle.cpp

```diff
@@ -1239,15 +1239,15 @@ void ViewportAddVehicles(DrawPixelInfo *dpi)
  * @param y  Y coordinate in the viewport.
  * @return Closest vehicle, or \c nullptr if none found.
  */
-Vehicle *CheckClickOnVehicle(const Viewport *vp, int x, int y)
+Vehicle *CheckClickOnVehicle(const Viewport &vp, int x, int y)
 {
 	Vehicle *found = nullptr;
 	uint dist, best_dist = UINT_MAX;
 
-	if ((uint)(x -= vp->left) >= (uint)vp->width || (uint)(y -= vp->top) >= (uint)vp->height) return nullptr;
+	if ((uint)(x -= vp.left) >= (uint)vp.width || (uint)(y -= vp.top) >= (uint)vp.height) return nullptr;
```

Comment:
`static_cast`?

Reply:
Switched to using splitting the subtraction out of the if conditions and and using `IsInsideMM()` (which does the casting itself.)